docs/user-guide/ingest-data/for-iot/grpc-sdks/java.md

```diff
@@ -62,379 +111,302 @@ String database = "public";
+    .addTag("region", DataType.String)
+    .addField("cpu_util", DataType.Float64)
+    .addField("memory_util", DataType.Float64)
+    .addTimestamp("ts", DataType.TimestampMillisecond)
+    .build();
+
+// Create a table from the schema
+Table table = Table.from(schema);
+
+// Add rows to the table
+// The values must be provided in the same order as defined in the schema
+// In this case: addRow(host, region, cpu_util, memory_util, ts)
+table.addRow("host1", "us-west-1", 0.42, 0.78, System.currentTimeMillis());
+table.addRow("host2", "us-west-2", 0.46, 0.66, System.currentTimeMillis());
+// Add more rows
+// ..
 
-You can set table options using the `Context`.
-For example, to set the `ttl` option, use the following code:
+// Complete the table to make it immutable. This finalizes the table for writing.
+// If users forget to call this method, it will automatically be called internally
+// before the table data is written.
+table.complete();
 
-```java
-Context ctx = Context.newDefault();
-// Add a hint to make the database create a table with the specified TTL (time-to-live)
-ctx = ctx.withHint("ttl", "3d");
-// Set the compression algorithm to Zstd.
-ctx = ctx.withCompression(Compression.Zstd)
-// Use the ctx when writing data to GreptimeDB
-// The data object `cpuMetric` and `memMetric` are described in the following sections
-CompletableFuture<Result<WriteOk, Err>> future = greptimeDB.write(Arrays.asList(cpuMetric, memMetric), WriteOp.Insert, ctx);
+// Write the table to the database
+CompletableFuture<Result<WriteOk, Err>> future = client.write(table);
 ```
 
-For how to write data to GreptimeDB, see the following sections.
+Insert data in JSON type
 
-## Low-level API
+GreptimeDB supports storing complex data structures using [JSON type data](/reference/sql/data-types.md#json-type). Here's an example of adding a JSON column:
 
-The GreptimeDB low-level API provides a straightforward method to write data to GreptimeDB 
-by adding rows to the table object with a predefined schema.
+```java
+// Construct the table schema for sensor_readings
+TableSchema sensorReadings = TableSchema.newBuilder("sensor_readings")
+        // The code for creating other columns is omitted
+        // ...
+        // specify the column type as JSON        
+        .addField("attributes", DataType.Json)
+        .build();
 
-### Create row objects
+// ...
+// Use map to insert JSON data
+Map<String, Object> attr = new HashMap<>();
+attr.put("location", "factory-1");
+sensorReadings.addRow(<other-column-values>... , attr);
+```
 
-This following code snippet begins by constructing a table named `cpu_metric`,
-which includes columns `host`, `cpu_user`, `cpu_sys`, and `ts`. 
-Subsequently, it inserts a single row into the table.
+##### TableSchema
 
-The table consists of three types of columns:
+The `TableSchema` defines the structure for writing data to GreptimeDB. It includes information about column names, semantic types, and data types.
```

Comment:
Most of this section is described in the data model documentation.
It is recommended to link to the data model documentation to reduce duplication.
Only left the unique content for Java SDK.